plugin/helpers_bots.go

```diff
@@ -124,6 +134,16 @@ func (p *HelpersImpl) ShouldProcessMessage(post *model.Post, options ...ShouldPr
 		option(messageProcessOptions)
 	}
 
+	botIdBytes, kvGetErr := p.API.KVGet(BOT_USER_KEY)
+	if kvGetErr != nil || botIdBytes == nil {
```

Comment:
Will `botIdBytes` be `nil` if no bot is being used?

Reply:
Yes it will be. The way I think we should handle this is **ignore** this self post and OnlyBotDM check if the plugin does not have any bot. Wdyt? Apologies this cases are not well thought through as I lack product understanding and haven't used different kinds of plugins. I did not know plugin without a bot can exists.

Reply:
Yes, plugins can exist without a bot. If `botIdBytes` we can just continue with the system message checks and skip the ones related to `botId`.